src/test/java/io/quarkus/search/app/SynonymSearchServiceTest.java

```diff
@@ -0,0 +1,231 @@
+package io.quarkus.search.app;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.when;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import io.quarkus.search.app.dto.GuideSearchHit;
+import io.quarkus.search.app.dto.SearchResult;
+import io.quarkus.search.app.quarkusio.QuarkusIO;
+import io.quarkus.search.app.testsupport.GitTestUtils;
+import io.quarkus.search.app.util.CloseableDirectory;
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.QuarkusTestProfile;
+import io.quarkus.test.junit.TestProfile;
+import io.restassured.RestAssured;
+import io.restassured.common.mapper.TypeRef;
+import io.restassured.filter.log.LogDetail;
+import org.apache.commons.io.file.PathUtils;
+import org.awaitility.Awaitility;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.eclipse.jgit.revwalk.RevCommit;
+
+@QuarkusTest
+@TestProfile(SynonymSearchServiceTest.Profile.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
```

Comment:
FWIW you should be able to use `@QuarkusIOSample.Setup(MyFilter.class)` to take your data from the actual QuarkusIO sample, using only the few files you're interested in.

That might be a better way to check that this fixes the problem?

Reply:
IDK, that was where I've started (with an annotation `@QuarkusIOSample.Setup(MyFilter.class)`) but then I thought that it may be simpler to have a "controlled" text to test the synonyms...  Let me give that one more try 😃